src/extension/features/accounts/bulk-edit-memo/index.jsx

```diff
@@ -5,57 +5,145 @@ import { l10n } from 'toolkit/extension/utils/toolkit';
 import { componentAfter } from 'toolkit/extension/utils/react';
 import { getEntityManager } from 'toolkit/extension/utils/ynab';
 
+const DEFAULT_DISPLAY_MODE = 'defaultDisplayMode';
+const MENU_DISPLAY_MODE = 'menuDisplayMode';
+const EDIT_DISPLAY_MODE = 'editDisplayMode';
+
+const REPLACE_EDIT_MODE = 'replaceEditMode';
+const ADD_PREFIX_EDIT_MODE = 'addPrefixEditMode';
+const ADD_SUFFIX_EDIT_MODE = 'addSuffixEditMode';
+
 const EditMemo = () => {
-  const [isEditMode, setIsEditMode] = useState(false);
+  const [displayMode, setDisplayMode] = useState(DEFAULT_DISPLAY_MODE);
+  const [editMode, setEditMode] = useState(REPLACE_EDIT_MODE);
   const [memoInputValue, setMemoInputValue] = useState('');
 
+  const makeNewMemo = (prevMemo) => {
+    switch (editMode) {
+      case REPLACE_EDIT_MODE:
+        return memoInputValue;
+      case ADD_PREFIX_EDIT_MODE:
+        return memoInputValue + prevMemo;
+      case ADD_SUFFIX_EDIT_MODE:
+        return prevMemo + memoInputValue;
+    }
+    return memoInputValue;
+  };
+
   const handleConfirm = () => {
     const checkedRows = containerLookup('service:accounts').areChecked;
     const { transactionsCollection } = getEntityManager();
     getEntityManager().performAsSingleChangeSet(() => {
       checkedRows.forEach((transaction) => {
         const entity = transactionsCollection.findItemByEntityId(transaction.get('entityId'));
+        const memoPrevValue = transaction.get('memo') || '';
         if (entity) {
-          entity.set('memo', memoInputValue);
+          entity.set('memo', makeNewMemo(memoPrevValue));
         }
       });
     });
 
     controllerLookup('accounts').send('closeModal');
+    setDisplayMode(DEFAULT_DISPLAY_MODE);
```

Comment:
I don't think the modal is closing anymore? this might be a different bug than from your PR might need to do `getAccountService.send('closeModal')`